### PR TITLE
feat: change Navigation selection UX and update UI to new design guidance

### LINKED ISSUE #2046: Add a selection state to the Navigation
The Navigation should have a selection state similar to that of the Navigation Menu, see #2035

### PR DESCRIPTION
**Pull request by janechu**
# Description

## Motivation & context

Updates to make the expand different from the selection of an item in Navigation and updates it to the new design.

Resolves #2046

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-tooling-react/src/navigation/navigation.style.ts [hunk 1]
```diff
@@ -101,12 +119,21 @@ const styles: ComponentStyles<NavigationClassNameContract, {}> = {
         cursor: "inherit",
         textDecoration: "none",
         fontStyle: "italic",
-        "&:focus": {
```
Comment (eljefe223) [inline packages/fast-tooling-react/src/navigation/navigation.style.ts]: I think we still want focus? Also should the expand collapse arrows have focus?

**Comment by eljefe223** [comment]
image

looks like we can get in a state where 2 elements appear selected. Also selecting the child element in this case collapses the node.

File: packages/fast-tooling-react/src/navigation/navigation.style.ts (hunk 1 above)
Comment (eljefe223) [inline packages/fast-tooling-react/src/navigation/navigation.style.ts]: I think we still want focus. Also should the expand collapse arrows have focus?

**Comment by janechu** [comment]
This is an existing issue and happens on non default react children, tracking now as #2082

File: packages/fast-tooling-react/src/navigation/navigation.style.ts (hunk 1 above)
Comment (janechu) [reply packages/fast-tooling-react/src/navigation/navigation.style.ts]: Added back in focus visible, the expand buttons shouldn't get focus (i believe)

**Pull request merged**